frontend/src/views/reports/SurveyView.vue

```diff
@@ -0,0 +1,497 @@
+
+    async back() {
+      if (this.isFirstSection(this.currentSection)) return
+      await this.save()
+      this.currentSection = this.sections[this.currentSectionIndex - 1]
+    },
+
+    toggleCancelDialog() {
+      this.showCancelDialog = !this.showCancelDialog
+    },
+
+    cancelChanges() {
+      this.clonedResponses = cloneDeep(this.originalResponses)
+      this.processQuestionsBusinessRules(this.currentSection)
+      this.toggleCancelDialog()
+      this.verifySectionComplete(this.currentSection)
+    },
+
+    async next() {
+      if (this.isLastSection(this.currentSection)) return
+      await this.save()
+      this.currentSection = this.sections[this.currentSectionIndex + 1]
+    },
+
+    async save(showAlert) {
+      try {
+        this.processing = true
+        const responsesToDelete = this.clonedResponses?.filter((response) => response.questionResponseId && this.isHiddenOrDeleted(response))
+        let callGetQuestionsResponses = responsesToDelete?.length > 0
+        await Promise.all(
+          responsesToDelete?.map(async (response) => {
+            const originalResponse = this.getOriginalQuestionResponse(response)
+            await ReportsService.deleteQuestionResponse(originalResponse?.questionResponseId)
+            originalResponse.rowId = null
+          }),
+        )
+        await Promise.all(
+          this.clonedResponses?.map(async (response) => {
+            if (this.isHiddenOrDeleted(response)) return
+            const originalResponse = this.getOriginalQuestionResponse(response)
+            response.value = Array.isArray(response.value) ? response.value?.toString() : response.value
+            if (isEmpty(originalResponse) && !isEmpty(response?.value)) {
+              await ReportsService.createQuestionResponse(response)
+              callGetQuestionsResponses = true
+            } else if (originalResponse?.value !== response?.value || originalResponse?.rowId !== response?.rowId) {
+              await ReportsService.updateQuestionResponse(originalResponse?.questionResponseId, response)
+              callGetQuestionsResponses = true
+            }
+          }),
+        )
+        if (callGetQuestionsResponses) {
+          await this.getQuestionsResponses()
+        }
+        this.verifySectionComplete(this.currentSection)
+        if (showAlert) {
+          this.setSuccessAlert('Survey response saved successfully')
+        }
+      } catch (error) {
+        this.setFailureAlert('Failed to save your survey responses', error)
+      } finally {
+        this.processing = false
+      }
+    },
+
+    async updateCurrentSection(section) {
+      if (this.loading || this.processing) return
+      await this.save()
+      this.currentSection = section
+    },
+
+    async submit() {
+      try {
+        const currentMonth = this.months?.find((month) => month.name === new Date().toLocaleString('en-ca', { month: 'long' }))
+        const payload = {
+          statusCode: 506580000,
```

Comment:
Does this have specific meaning in CRM? If so, should it be defined as a constant?

Reply:
it's updated